File: agentplus/app/src/main/java/merryweather/com/agentplus/di/modules/RepositoryModule.java
package merryweather.com.agentplus.di.modules;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import merryweather.com.agentplus.repository.AgentplusRepository;

/**
 * Created by S on 19.05.2018.
 */

@Module
public class RepositoryModule {

    @Singleton
    @Provides
    public AgentplusRepository provideHarassmentRepository() {
        return new AgentplusRepository();
    }
}

File: Nista/app/src/main/java/com/example/nista/di/modules/RepositoryModule.kt
package com.leroymerlin.cds.di.di.modules

import android.content.Context
import com.example.nista.api.MovieApi
import com.example.nista.api.MovieService
import com.example.nista.repository.MovieRepository
import com.leroymerlin.cds.di.qualifier.AppContext

import javax.inject.Singleton

import dagger.Module
import dagger.Provides



@Module
class RepositoryModule {

    @Provides
    @Singleton
    fun provideMovieRepository(service : MovieService, @AppContext context : Context) : MovieRepository {
        return MovieRepository(service)
    }

}
